test/unittest/dynamic/discovery_database/DiscoveryDatabaseTest.cpp

```diff
@@ -0,0 +1,208 @@
+// Copyright 2021 Proyectos y Sistemas de Mantenimiento SL (eProsima).
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#include <gtest_aux.hpp>
+#include <gtest/gtest.h>
+
+#include <ddsrouter/dynamic/DiscoveryDatabase.hpp>
+#include <ddsrouter/exceptions/InconsistencyException.hpp>
+#include <ddsrouter/types/endpoint/Endpoint.hpp>
+#include <ddsrouter/types/endpoint/Guid.hpp>
+#include <ddsrouter/types/endpoint/QoS.hpp>
+#include <ddsrouter/types/ReturnCode.hpp>
+#include <ddsrouter/types/topic/RealTopic.hpp>
+
+using namespace eprosima::ddsrouter;
+
+Guid random_guid(
+        uint16_t seed = 1)
+{
+    Guid guid;
+    guid.entityId.value[3] = seed;
+    guid.guidPrefix.value[0] = 0x01;
+    guid.guidPrefix.value[1] = 0x0f;
+    return guid;
+}
+
+/**
+ * Test \c DiscoveryDatabase \c topic_exists method
+ *
+ * CASES:
+ *  Topic not present in database
+ *  Topic present in database
+ */
+TEST(DiscoveryDatabaseTest, topic_exists)
+{
+    DiscoveryDatabase discovery_database;
+    Guid guid;
+    QoS qos;
+    RealTopic topic("test", "test");
+    Endpoint endpoint(EndpointKind::READER, guid, qos, topic);
+
+    ASSERT_FALSE(discovery_database.topic_exists(topic));
+    discovery_database.add_endpoint(endpoint);
+    ASSERT_TRUE(discovery_database.topic_exists(topic));
+}
```

Comment:
Please, add a more complex test by creating two endpoints in same topic, and erasing one must result in the topic still existing.

Reply:
:+1: